imaginairy/schema.py

```diff
@@ -172,385 +133,543 @@ def handle_b64(value: Any) -> "LazyLoadingImage":
+            prompt_strength=prompt_strength,
+            init_image=init_image,
+            width=width,
+            height=height,
+            seed=seed,
+            steps=steps,
+            model=model,
+            sampler_type=sampler_type,
+            control_inputs=control_inputs,
+            outpaint=outpaint,
+            face_fixer=face_fixer,
+            tiling=tiling,
+            caption_text=caption_text,
+        )
+        self.__dict__.update(self._model)
 
     @property
-    def prompt_text(self):
-        if not self.prompt:
-            return ""
+    def prompt_text(self) -> str:
         if len(self.prompt) == 1:
             return self.prompt[0].text
         return "|".join(str(p) for p in self.prompt)
 
     @property
-    def negative_prompt_text(self):
-        if not self.negative_prompt:
-            return ""
+    def negative_prompt_text(self) -> str:
         if len(self.negative_prompt) == 1:
             return self.negative_prompt[0].text
         return "|".join(str(p) for p in self.negative_prompt)
 
-    def prompt_description(self):
+    @property
+    def init_image_strength(self) -> float:
+        return self.init_image.strength if self.init_image else 0
+
+    def prompt_description(self) -> str:
         return (
             f'"{self.prompt_text}" {self.width}x{self.height}px '
             f'negative-prompt:"{self.negative_prompt_text}" '
             f"seed:{self.seed} prompt-strength:{self.prompt_strength} steps:{self.steps} sampler-type:{self.sampler_type} init-image-strength:{self.init_image_strength} model:{self.model}"
         )
 
-    def logging_dict(self):
-        """Return a dict of the object but with binary data replaced with reprs."""
-        data = self.model_dump()
-        data["init_image"] = repr(self.init_image)
-        data["mask_image"] = repr(self.mask_image)
-        if self.control_inputs:
-            data["control_inputs"] = [repr(ci) for ci in self.control_inputs]
+    def full_copy(self, deep: bool = True, update: dict[str, Any] | None = None):
+        """Return a copy of the object while perfoming an update."""
+        if update is None:
+            update = {}
+        data = copy.deepcopy(self) if deep else copy.copy(self)
+        data.__dict__.update(update)
         return data
 
-    def full_copy(self, deep=True, update=None):
-        new_prompt = self.model_copy(
-            deep=deep,
-            update=update,
-        )
-        # new_prompt = self.model_validate(new_prompt) doesn't work for some reason https://github.com/pydantic/pydantic/issues/7387
-        new_prompt = new_prompt.model_validate(dict(new_prompt))
-        return new_prompt
-
     def make_concrete_copy(self):
-        seed = self.seed if self.seed is not None else random.randint(1, 1_000_000_000)
-        return self.full_copy(
-            deep=False,
-            update={
-                "seed": seed,
-            },
-        )
+        return self.full_copy(deep=True)
```

Comment:
There was some reason this did not work. I should have left a comment because I dont remember.

Reply:
Yes, but now it's a plain python dataclass and not a pydantic model ; I'll do some testing to verify it works correctly